### DISCUSSION TITLE
Recommended Local Storage for Offline Support

### DISCUSSION THREAD

**Discussion by divinemaredi** [supabase/supabase#1271] [Questions]
Hi there!

I was happy to see here #357 that Supabase will have offline support since I'm developing a flutter app that will need it in the future.

With Offline Support in mind, I wanted to know if it's recommended to develop an app that uses an SQL database for local storage (e.g. [sqflite](https://pub.dev/packages/sqflite))  if that will make it easier in the future to integrate Offline Support since Supabase uses Postgres.

Or will using a No-SQL db for local storage (e.g. [hive](https://pub.dev/packages/hive)) make no difference with regards to integrating Offline Support in the future?

Thanks!

**Comment by silentworks** [comment]
I have no idea what is planned for their offline storage but one method to help work around this is to create an abstraction on top of whichever system you will use for offline storage, that way you can have the same API when Supabase is ready and just change the underlying implementation.

**Reply by divinemaredi** [reply]
Thanks @silentworks, that's a solid tip!
